apis/network/v1beta1/ipset_webhook.go

```diff
@@ -74,3 +134,134 @@ func (r *IPSet) ValidateDelete() error {
+func valiateIPSetNetwork(
+	networks []IPSetNetwork,
+	path *field.Path,
+	netCfgSpec *NetConfigSpec,
+) field.ErrorList {
+	allErrs := field.ErrorList{}
+	netNames := map[string]field.Path{}
+	defaultRouteCount := map[k8snet.IPFamily]int{
+		k8snet.IPv4:            0,
+		k8snet.IPv6:            0,
+		k8snet.IPFamilyUnknown: 0, // should not be required since the cidr gets validated on the NetConfig
+	}
+
+	for _netIdx, _net := range networks {
+		path := path.Child("networks").Index(_netIdx)
+
+		// validate uniqe networks in request
+		allErrs = append(allErrs, valiateUniqElement(netNames, string(_net.Name), path, "name", errDupeNetworkName)...)
+
+		// validate if requested network is in NetConfig
+		f := func(n Network) bool {
+			return n.Name == _net.Name
+		}
+		netIdx := slices.IndexFunc(netCfgSpec.Networks, f)
+
+		if netIdx >= 0 {
+			// validate if requested subnet is in NetConfig
+			f := func(n Subnet) bool {
+				return n.Name == _net.SubnetName
+			}
+			subNetIdx := slices.IndexFunc(netCfgSpec.Networks[netIdx].Subnets, f)
+
+			if subNetIdx >= 0 {
+				// net and subnet are valid
+
+				if _net.FixedIP != nil || _net.DefaultRoute != nil {
+					cidr := netCfgSpec.Networks[netIdx].Subnets[subNetIdx].Cidr
+					_, ipPrefix, ipPrefixErr := net.ParseCIDR(cidr)
+					if ipPrefixErr != nil {
+						// this should never happen as the subnet CIDR was already validated
+						// via the netcfg webhook
+						allErrs = append(allErrs, field.Invalid(path.Child("cidr"), cidr, errInvalidCidr))
+						return allErrs
+					}
+					ipFam := k8snet.IPFamilyOfCIDR(ipPrefix)
+
+					// validate the requested FixedIP
+					if _net.FixedIP != nil {
+						path := path.Child("fixedIP")
+						if err := valiateAddress(*_net.FixedIP, ipPrefix, path); err != nil {
+							allErrs = append(allErrs, err...)
+						}
+					}
+
+					// check that there are not multiple have the defaultRoute flag
+					if _net.DefaultRoute != nil && *_net.DefaultRoute {
+						defaultRouteCount[ipFam]++
+
+						for fam, count := range defaultRouteCount {
+							if count > 1 {
+								allErrs = append(allErrs, field.Invalid(path.Child("defaultRoute"), _net.Name, fmt.Sprintf(errMultiDefaultRoute, string(fam))))
+							}
+						}
+					}
+				}
+
+				continue
+			}
+
+			// the requested subnet was not found in the network
+			allErrs = append(allErrs, field.Invalid(path.Child("subnetName"), _net.SubnetName, fmt.Sprintf(errSubnetNotInNetwork, _net.SubnetName, _net.Name)))
+			continue
+		}
+
+		// the requested net was not found in the NetConfig
+		allErrs = append(allErrs, field.Invalid(path.Child("name"), _net.Name, fmt.Sprintf(errNetworkNotFound, _net.Name)))
+	}
+
+	return allErrs
+}
+
+// valiateIPSetChanged
+// - if a previous requested network is still in the list
+// - if subnet changed within a network
+// - if fixedIP changed
+// - if defaultRoute changed
+func valiateIPSetChanged(
+	networks []IPSetNetwork,
```

Comment:
Can we not mark IPSet CRD immutable with the following kubebuilder comment and won't need to implement things in webhook for update? 

// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"

Reply:
haven't check `XValidation`, will look into that. in general how about adding a new network to a compute node or a service? shouldn't this be possible?

how about this approach,  we add an immutable flag to the ipset and if set block any update, if not its up to the client? for a service on OCP which uses ipset it should be ok to update it, as it just gets restarted with the new settings.

Reply:
> how about adding a new network to a compute node?

I've not seen cases of customers adding new networks on existing compute nodes, though it could be a valid use-case.

> how about this approach, we add an immutable flag to the ipset and if set block any update, if not its up to the client?

With the amount of support cases that we've seen where customer was not using fixed/predictable IP and IP changes causing chaos, I think making it immutable/adding an immutable flag would probably be better. I guess we would also need to add finalizers on IPSets from role/node. I did not add them atm because there is not reconcile delete logic for node/role atm for scale-down.

Reply:
ack, let me add that immutable flag, which the client then can set and we can set it from the dataplane-operator.

Reply:
@rabi @abays this is now added in a separate commit. Might make review easier. `Spec.Immutable` on the IPSet can be set to block spec updates.
I was wondering if it make sense to still be able to flip the Immutable for future fixing of envs. means if Immutable set you can not update the rest of the spec, but if required you have to explicitly update the object flip the Immutable and then could do changes. but can do that in a follow up if you think that  makes sense.

Reply:
I've seen adding new networks a few times, for example to enable use of some new NFS or iSCSI storage backend they need to add VLAN/interface. I like the `Immutable` approach - forcing an additional step to allow changes makes sense and sends a good "be careful" signal.